src/explore-education-statistics-common/src/services/permalinkSnapshotService.ts

```diff
@@ -40,25 +40,28 @@ export interface PermalinkSnapshot {
 }
 
 interface CreatePermalink {
-  query: TableDataQuery;
+  query: ReleaseTableDataQuery;
   configuration: {
     tableHeaders: UnmappedTableHeadersConfig;
   };
 }
 
 const permalinkSnapshotService = {
-  createPermalink(query: CreatePermalink): Promise<PermalinkSnapshot> {
-    return dataApi.post(`/permalink-snapshot`, query);
+  createPermalink(permalink: CreatePermalink): Promise<PermalinkSnapshot> {
+    const { releaseId } = permalink.query;
+    return releaseId
+      ? dataApi.post(`/permalink/release/${releaseId}`, permalink)
+      : dataApi.post('/permalink', permalink);
```

Comment:
Not super necessary to change this in the hotfix, but not sure why we opted for a separate endpoints to attach a release?

We could just have a single `POST /api/permalink` endpoint which takes an optional `releaseId` in the request body. This would probably allow us to simplify the service code to a single method as well as part of this.

Would change this as part of this PR if it's not too much trouble, or consider changing it in a non-hotfix PR at least.

Reply:
I'm not sure why that's the case, it's been that way since release amendments were introduced.

I initially thought about making this change but decided not to because ideally I wanted to combine it with making a  change to the `TableBuilderService.Query` to have optional `releaseId` which would clean things up by avoiding having to working out the release id in `PermalinkService`.

I'd also considered using `ReleaseSubjectService.Find` to find the ReleaseSubject from an optional releaseId, but looking a the code it doesn't quite have the same behaviour as the way we determine the releaseId in `PermalinkService` and `TableBuilderService`.

I've made the change you suggested on this PR now but still work out `releaseId` in `PermalinkService`. That's needed because we store set it on the permalink entity that's saved to the database for reference.